movienager/accounts/models.py

```diff
@@ -39,6 +39,9 @@ class Meta:
 
     def save(self, *args, **kwargs):
         super().save(*args, **kwargs)
+        self.set_permission()
+
+    def set_permission(self):
```

Comment:
What about:
```
    def set_permission(self):
        if self.user_type == self.ADMIN:
            self.add_manage_permission(self)
        else:
            self.remove_manage_permission(self)

    @classmethod
    def get_permission(cls, user, perm_codename):
        content_type = ContentType.objects.get_for_model(cls)
        return Permission.objects.get(
            codename=perm_codename, content_type=content_type
        )

    def add_manage_permission(self, user):
        permission = self.get_permission(self, user, "manage user")
        user.user_permissions.add(permission)

    def remove_manage_permission(self, user):
        permission = self.get_permission(self, user, "manage user")
        user.user_permissions.remove(permission)
```
Function composition over conditional flow control

Reply:
You can also use built-in permissions like `accounts.add_user`. You can use string literals to directly add or remove built-in perms. See: https://docs.djangoproject.com/en/3.0/topics/auth/default/#default-permissions

>The Permission model is rarely accessed directly. ⬆️

Reply:
I get the following error "invalid literal for int() with base 10: 'accounts.user_add'".

Reply:
I see, probably you have to pass Permission object, so we have to live with that. Sorry for misleading comment